models/articles_models.js

```diff
@@ -46,20 +46,64 @@ exports.patchArticleById = (inc_votes, article_id) => {
     });
 };
 
-exports.fetchArticles = () => {
-  return connection
-    .query(
-      `SELECT articles.*, COUNT(comments.article_id)::INT AS comment_count 
-    FROM articles 
-    LEFT JOIN comments 
-    ON comments.article_id = articles.article_id
-    GROUP BY articles.article_id
-    ORDER BY created_at DESC`
-    )
-    .then((articles) => {
-      return articles.rows;
-    })
-    .catch((err) => {
-      return err;
+exports.fetchArticles = async (
+  sort_by = "created_at",
+  order = "desc",
+  topic
+) => {
+  const validSort_by = [
+    "topic",
+    "title",
+    "author",
+    "body",
+    "created_at",
+    "votes",
+  ];
+
+  const validOrder = ["ASC", "DESC", "asc", "desc"];
```

Comment:
this could probably be tidied up by using a `.toLowerCase()` or `.toUpperCase()` somewhere